### PR TITLE
Add Notify error alert on editor validation error

### LINKED ISSUE #1307: Add notify alert on form validation error

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #1307

### CODE REVIEW

**Comment by taylortom** [comment]
Some quite major changes here, so needs to be tested thoroughly.

**Comment by dancgray** [comment]
+1 works for me. Is it possible to return the attribute title in the error message? e.g "Scroll Duration" rather than _scrollDuration (for trickle).

**Comment by dancgray** [comment]
Ahh, the page edit sidebar is not being updated on close of the alert error

screen shot 2016-08-11 at 09 39 48

**Comment by taylortom** [comment]
Hmm...may need to call the parent method

File: frontend/src/core/editor/config/views/editorConfigEditView.js [hunk 1]
```diff
@@ -1,60 +1,29 @@
...
-          Origin.trigger('editingOverlay:views:hide');
-
-          Origin.trigger('editor:refreshData', function() {
-            Backbone.history.history.back();
-            this.remove();
-          }, this);
-
-        }, this)
-      });
+      return _.extend(changed, mandatory);
```
Comment (tomgreenfield) [inline frontend/src/core/editor/config/views/editorConfigEditView.js]: `mandatory` isn't defined here.

File: frontend/src/core/editor/page/views/editorComponentEditView.js [hunk 2]
```diff
@@ -21,49 +18,23 @@ define(function(require) {
...
-      // This must trigger no matter what, as sidebar needs to know
-      // when the form has been resubmitted
-      Origin.trigger('editorSidebar:showErrors', errors);
-      if (errors) {
-        return;
-      }
-      self.form.commit();
-
+    getAttributesToSave: function() {
       self.model.set('_componentType', self.model.get('_componentType')._id);
```
Comment (tomgreenfield) [inline frontend/src/core/editor/page/views/editorComponentEditView.js]: Undefined `self.model` is causing me issues.

**Comment by taylortom** [comment]
@dancgray we don't get any more data than the attribute name and a message. Could maybe try and look this up somehow

**Comment by taylortom** [comment]
Have tweaked form.validate to also pass the field title

File: frontend/src/core/editor/page/views/editorComponentEditView.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ define(function(require) {
         this.remove();
       }, this);
     },
+    */
```
Comment (tomgreenfield) [inline frontend/src/core/editor/page/views/editorComponentEditView.js]: Has this crept in by mistake?

**Comment by tomgreenfield** [comment]
+1

**Comment by lc-thomasberger** [comment]
in project setting, when title is empty the error message is generic. Is it possible to add title is required - works for components and blocks?

**Comment by tomgreenfield** [comment]
@lc-thomasberger, good spot.

[`projectDetailView.js`](https://github.com/adaptlearning/adapt_authoring/blob/master/frontend/src/core/project/views/projectDetailView.js) will need your magic, @taylortom.

**Comment by taylortom** [comment]
@lc-thomasberger, @tomgreenfield fixed

**Comment by tomgreenfield** [comment]
+1

**Comment by dancgray** [comment]
+1

**Comment by tomgreenfield** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

**Pull request merged**